### ISSUE TITLE
Next Js -  Hot reloading occurs on every save when having badly defined content in tailwing config file.

### ISSUE THREAD

**Issue by JustCreateDeveloper** [tailwindlabs/tailwindcss#14555]
**What version of Tailwind CSS are you using?**

3.4.13

**What build tool (or framework if it abstracts the build tool) are you using?**
NextJs 14.2.2

**What version of Node.js are you using?**

21.3.0

**What browser are you using?**

Chrome, Opera

**What operating system are you using?**

macOS, Windows

**Reproduction URL**

https://github.com/vercel/next.js/discussions/25712#discussioncomment-10354430

**Describe your issue**

Hot reloading occurs on every save when having badly defined content in tailwing config file.

**Comment by RobinMalfait** [comment]
Hey!

When you setup the `content` option in a way where it matches _every_ file (or unexpected files), then saving any of the files it matches will trigger a rebuild and a hot reload will likely happen as well. 

If I understand the problem correctly, it means that you were listening for changes in the `src/` directory instead of the `src/app/` directory.

Since you are telling Tailwind to look for changes in all of `src/`, it's impossible to know that this was wrong from Tailwind's perspective. You asked it to watch in that folder, if we then see changes an update will happen.

If you can elaborate a bit more about what your issue was, and how we can improve things then I'm all ears!

Long story short, this feature currently works as designed. Going to close this issue for this reason.

The part I don't understand and am curious about is:
1. What is the issue with hot reloading exactly? If a change happened in `src/` somewhere, a hot reload could be triggered, but I'm not sure if that is an issue or not. Did something break?
1. What files exist in `src/` that are siblings of the `app/` directory that caused fast refresh issues? Were those files that were generated by something, or are you manually changing those files?
